### PR TITLE
Add support for fish completion

### PR DESCRIPTION
**Pull request by sayboras**
Add completion support for fish shell

image

### CODE REVIEW

File: go.mod
Comment (sayboras) [inline go.mod:66]: The related PR is pending for review and most likely to be merged soon. Then I will remove this replace directive statement.

File: go.mod
Comment (sayboras) [inline go.mod:46]: this is the from the master branch (after related PR is merged), we can wait for official release as well.

Comment (SVilgelm) [reply go.mod:46]: I asked @spf13 about the release date, hopefully he will answer
https://gophers.slack.com/archives/CD3LP1199/p1594518169033000

Comment (SVilgelm) [reply go.mod:46]: So if we wait the cobra release, then could you mark it as WIP?

Comment (SVilgelm) [reply go.mod:46]: or we can merge

Comment (sayboras) [reply go.mod:46]: Oh I am not sure what the practice we follow here in this project, so keen to hear inputs from other you have more experience working on this :). Marking this as draft will not pop up notification.

/cc @golangci/core-team

Comment (SVilgelm) [reply go.mod:46]: you can add `WIP` prefix in title, then the WIP check will block this PR from merging

Comment (sayboras) [reply go.mod:46]: cool hope that team will still check WIP PR, I personally don't :laughing:

**Comment by sayboras** [comment]
@golangci/team seems like cobra might take time to release tagged version, just wanna check if you guys are ok with git hash version from master. If yes, I can merge this in, otherwise, will mark this as draft.

**Review by denis-tingajkin — APPROVED** [review]

File: pkg/commands/completion.go
Comment (saschagrunert) [inline pkg/commands/completion.go:65]:
```suggestion
		return errors.Wrap(err, "generate fish completion")
```

**Review by saschagrunert — APPROVED** [review]

**Comment by Alwayswithme** [comment]
@sayboras hi, I generate completion script  at directory `golangci-lint completion fish > ~/.config/fish/completions/`

but encouter issue when `golangci-lint run <tab>`, can u teach me how to setup completion for fish?
```sh
➜  completions golangci-lint run math: Error: Unexpected token
'ompletion ended with directive: ShellCompDirectiveDefault / 1'
         ^
math: Error: Unexpected token
'% 2'
 ^
test: Missing argument at index 2

~/.config/fish/completions/golangci-lint.fish (line 99): 
    if test $compErr -eq 1
       ^
in function '__golangci_lint_prepare_completions'
in command substitution

(Type 'help test' for related documentation)
math: Error: Unexpected token
'ompletion ended with directive: ShellCompDirectiveDefault / 8'
         ^
math: Error: Unexpected token
'% 2'
 ^
```

golangci-lint has version 1.33.0 built from b90551c on 2020-11-23T06:54:06Z
fish, version 3.1.2

**Comment by sayboras** [comment]
have you try with `~/.config/fish/completions/golangci-lint.fish` ? 

```
tam.mach@24430 ~/g/s/g> golangci-lint
cache  (Cache control and information)  help                                                                                                (Help)  version  (Version)
completion  (Output completion script)  linters                                                               (List current linters configuration)
config                        (Config)  run  (Run this tool in cloud on every github pull request in https://golangci.com for free (public repos))

tam.mach@24430 ~/g/s/g> golangci-lint version
golangci-lint has version 1.33.0 built from b90551c on 2020-11-23T06:54:06Z
```

**Pull request merged**
